### PR TITLE
DAOS-4223 snapshot: Propagate snap IV after commit

### PR DESCRIPTION
**Pull request by NiuYawei**
On create/destroy snapshots, the snapshot IV update needs be called
after rdb tx commit, otherwise, new change to the snapshots won't
be fetched and aggregation won't be informed with uptodate snapshot
information.

Changed some aggregation related debug messages to use DB_EPC (which
are widely used for aggregation code).

Signed-off-by: Niu Yawei <yawei.niu@intel.com>

### CODE REVIEW

**Comment by NiuYawei** [comment]
Tests failed for DAOS-3945 and DAOS-4025. Liwei, Di, could you review this when you have time? Thanks.

File: src/container/srv_container.c [hunk 1]
```diff
@@ -2082,13 +2082,21 @@ cont_op_with_svc(struct ds_pool_hdl *pool_hdl, struct cont_svc *svc,
...
 	if (rc != 0)
 		D_GOTO(out_lock, rc);
 
 	rc = rdb_tx_commit(&tx);
+	if (rc != 0)
+		D_GOTO(out_lock, rc);
+
+	if (opc == CONT_SNAP_CREATE || opc == CONT_SNAP_DESTROY) {
+		D_ASSERT(cont != NULL);
+		ds_update_snap_iv(&tx, cont);
```
Comment (liw) [inline src/container/srv_container.c]: Using tx after committing it is considered hacky; the expected way is to read the previous list of snapshots and add/remove the new/victim snapshot.

Comment (NiuYawei) [reply src/container/srv_container.c]: What about starting another tx to read the snapshot list?

Comment (liw) [reply src/container/srv_container.c]: Using another TX would be acceptable, though a bit less preferable than the expected way.

Comment (wangdi1) [reply src/container/srv_container.c]: ah, so update_snap_iv() is actually using the original snapshots?  Maybe you can fix this function?

Comment (NiuYawei) [reply src/container/srv_container.c]: I discussed with Liwei, we agreed on that we can compromise to begin another tx to read and propagate snap list.

I updated the patch (the patch is squashed and force pushed), please review it again. Thanks.

File: src/container/srv_epoch.c [hunk 2]
```diff
@@ -506,3 +469,61 @@ ds_cont_get_snapshots(uuid_t pool_uuid, uuid_t cont_uuid,
 
 	return rc;
 }
+
+/*
+ * Propagate new snapshot list to all servers through snapshot IV, errors
+ * are ignored.
+ */
+void
+ds_cont_update_snap_iv(struct ds_pool *pool, uuid_t cont_uuid)
```
Comment (liw) [inline src/container/srv_epoch.c]: Could the pool parameter be replaced with a svc parameter? The IV NS is available with svc->cs_pool->sp_iv_ns. The pool UUID is available with svc->cs_pool_uuid.

Comment (NiuYawei) [reply src/container/srv_epoch.c]: Sure, thanks.

**Pull request merged**